sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -209,4 +277,45 @@ protected void sleepIfRunningAgainstService(long millis) {
             throw logger.logExceptionAsWarning(new IllegalStateException(ex));
         }
     }
+
+    /**
+     * Get offset which determines the rolling strategy.
+     *
+     * @return The offset according to day of the week and platform information
+     */
+    private static int getOffset() {
+        if (testMode == TestMode.PLAYBACK) {
+            return 0;
+        }
+        LocalDate today = LocalDate.now(ZoneId.systemDefault());
+        return (today.getDayOfWeek().getValue() + getPlatFormOffset()) % PLATFORM_COUNT;
+    }
+
+    private static Integer getPlatFormOffset() {
+        String currentOs = System.getProperty("os.name").toLowerCase(Locale.ROOT);
+        System.out.println("It is currently running on os: " + currentOs);
+        String currentJdk = System.getProperty("java.version").toLowerCase(Locale.ROOT);
+        System.out.println("It is using jdk: " + currentJdk);
```

Comment:
Be sure to remove a lot of these println's before merging.

Reply:
Srikanta recommends to have the currentOs and currentJdk printout for debugging.

Reply:
Is it possible to use logger instead of S.o.p? That will allow us to turn on/off the logs.

Reply:
Will do.